fix parameter missing  when plugin type is a funtion

xiaoxiaojx opened the pull request:
> Hello, here's a small change ...

/lib/webpack.js (now)
```
  if (typeof plugin === "function") {
    plugin.apply(compiler);  // missing parameter
  } else {
    plugin.apply(compiler);
  }
```
/lib/webpack.js (after)
```
  if (typeof plugin === "function") {
    plugin.apply(null, [compiler]); // yes
  } else {
    plugin.apply(compiler);
  }
```
/declarations/WebpackOptions.d.ts
```
type WebpackPluginFunction = (
	compiler: import("../lib/Compiler")
) => void;
```

**What kind of change does this PR introduce?**

a bugfix

ooflorent commented:
Okay. I investigated a bit. This change is wrong. We want `this` to be the `compiler`.
`plugin.call(compiler, compiler)` may be better bacause it prevents confusion about `apply` and would work with `this` and arrow functions accepting the compiler as the first parameter. cc @sokra

sokra commented:
yeah makes sense. It also seems like the typings are wrong here. It should be:

``` typescript
type WebpackPluginFunction = (
	this: import("../lib/Compiler")
) => void;
```

resp. after the change:


``` typescript
type WebpackPluginFunction = (
	this: import("../lib/Compiler"),
	compiler: import("../lib/Compiler")
) => void;
```

xiaoxiaojx commented:
```
plugin.call(compiler, compiler)

type WebpackPluginFunction = (
	this: import("../lib/Compiler"),
	compiler: import("../lib/Compiler")
) => void;
``` 
>  😁 haha.. That's what I thought before. 
I changed the code. This should be a more compatible solution.

sokra commented:
> declarations/WebpackOptions.d.ts need to be updated

run `yarn special-lint-fix`

xiaoxiaojx commented:
> declarations/WebpackOptions.d.ts need to be updated

run `yarn special-lint-fix`

-----------------------------------------------------------------
Oh, I changed /declarations/WebpackOptions.d.ts directly before.
now, I changed schemas/WebpackOptions.json.
and run `yarn special-lint-fix`，hope this change is OK

sokra commented:
Thanks
